### ISSUE TITLE
Marker setPosition not reflected in map display

### ISSUE THREAD

**Issue by jobsonm** [tapioaali/GoogleMapsVaadin7#43]
I've been thinking about upgrading my plugin from 1.0.2 to the latest version but in doing so the marker setPosition doesn't seem to relocate any existing markers.

I'm running in a military training environment whereby there are a lot of entities moving at any one time (fast moving aircraft included). I receive the positional data from all the entities and convert these into Marker objects.

When I receive an update entity position I can setPosition with the new position of the aircraft. This functionality worked fine up to and including version 1.0.2 of this plugin. Post that version the setPosition call has no effect on the location of the Marker object.

Just test with the latest version (v1.3.4) and the effect (or lack of!) is the same. I can create the markers correctly and they appear in the correct location. But they do not move as I updated them with the setPostion call.

Being an extremely useful additional to my app though. Many thanks

Mike

**Comment by changty** [comment]
I'm also experiencing similar issues. I'm new to using google maps add-on, so I have no previous experience of working versions.

I've also noticed that changing the icon of the marker does not update on the map. I even tried marking the map dirty.

**Comment by tapioaali** [comment]
Verified. Quite weird that I haven't noticed this before since the issue is visible also in the demo: 'Move Kakola marker' button is not working.

I guess it's the equals method that compares just the ids of the markers and thus no changes are noticed. I modified the add-on to work like this when I did a major rewrite. That's why older versions work correctly but newer ones don't.

I'll definitely fix this to the next version.
